en/tutorials/module_template.md

```diff
@@ -0,0 +1,24 @@
+# Module Template
+
+The PX4 Firmware contains a template for writing a new application, which can be
+found under
+[src/templates/module](https://github.com/PX4/Firmware/tree/master/src/templates/module).
+It can be used for writing an application that runs on its own
+stack (as opposed to a work-queue task. This will be explained in more detail on
+the [Architecture Overview page](../concept/architecture.md#runtime-environment)).
+
+The template includes various aspects that might be useful for a full
+application:
+
+- Accessing parameters and reacting to parameter updates.
+- uORB subscriptions and waiting for topic updates.
+- Controlling the task that runs in the background via `start`/`stop`/`status`.
+  The `module start [<arguments>]` command can then be directly added to the
+  startup script.
+- Command-line argument parsing.
+- Documentation: the `PRINT_MODULE_*` methods serve two purposes:
+  - They are used to print the command-line usage when entering `module help` on the console.
+  - They are automatically extracted via script to generate the
```

Comment:
Is there nothing else that needs to be done - ANY module with these strings will have the commands reference generated?

Reply:
... probably even if it is true, we should say where it is generated, and note that it still needs to be copied into the library and TOC needs to be updated.